cdap-api-hadoop/src/main/java/co/cask/cdap/mapreduce/AbstractMapper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.mapreduce;
+
+import co.cask.cdap.api.ProgramLifecycle;
+import com.google.common.io.Closeables;
+import org.apache.hadoop.mapreduce.Mapper;
+
+import java.io.IOException;
+
+/**
+ * Abstract Mapper class which provides functionality to a MapReduceTaskContext object, which allows writing to
+ * multiple outputs.
+ *
+ * @param <KEYIN>    input key type
+ * @param <VALUEIN>  input value type
+ * @param <KEYOUT>   output key type
+ * @param <VALUEOUT> output value type
+ */
+public abstract class AbstractMapper<KEYIN, VALUEIN, KEYOUT, VALUEOUT>
+  extends Mapper<KEYIN, VALUEIN, KEYOUT, VALUEOUT>
+  implements ProgramLifecycle<MapReduceTaskContext> {
+
+  private MapReduceTaskContext<KEYIN, VALUEIN, KEYOUT, VALUEOUT> context;
+
+  @Override
+  public void initialize(MapReduceTaskContext context) throws Exception {
+    this.context = context;
+  }
+
+  @Override
+  public void destroy() {
+    Closeables.closeQuietly(context);
```

Comment:
What close the context here? In general, you only release resource that you created. Since context is passed to this class, it shouldn't be closed in here. It should be the call (which I believe is the MapperWrapper) to do the cleanup.